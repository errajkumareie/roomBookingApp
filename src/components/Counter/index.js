import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {faPlusCircle, faMinusCircle} from '@fortawesome/free-solid-svg-icons';
import styles from '../../styles/styles.js';

const Counter = props => {
  return (
    <View style={styles.innerContainer}>
      <TouchableOpacity
        disabled={props.decrementDisabled}
        activeOpacity={0.9}
        onPress={props.onDecrementPressed}>
        <FontAwesomeIcon icon={faMinusCircle} style={styles.icon} />
      </TouchableOpacity>
      <Text style={styles.counterText}>{props.count}</Text>
      <TouchableOpacity
        disabled={props.incrementDisabled}
        activeOpacity={0.9}
        onPress={props.onIncrementPressed}>
        <FontAwesomeIcon icon={faPlusCircle} style={styles.plusIcon} />
      </TouchableOpacity>
    </View>
  );
};

export default Counter;
