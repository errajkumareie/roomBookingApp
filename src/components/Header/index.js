import React from 'react';
import {Text, View} from 'react-native';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import styles from '../../styles/styles.js';

const Header = props => {
  return (
    <View style={styles.innerContainer}>
      <FontAwesomeIcon icon={props.source} style={styles.icon} />
      <Text style={styles.subTitle}>{props.name}</Text>
    </View>
  );
};
export default Header;
