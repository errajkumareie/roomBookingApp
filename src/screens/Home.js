import React, {useState} from 'react';
import {Text, View} from 'react-native';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {
  faUsers,
  faBed,
  faUser,
  faChild,
} from '@fortawesome/free-solid-svg-icons';
import Header from '../components/Header';
import Counter from '../components/Counter';
import styles from '../styles/styles.js';

const Home = () => {
  const [roomCount, setRoomCount] = useState(1);
  const [adultsCount, setAdultsCount] = useState(1);
  const [childrenCount, setChildrenCount] = useState(0);
  const minLimit = 1;
  const maxLimit = 4;
  const maxRoomLimit = 5;
  const totalCount = adultsCount + childrenCount;
  const maxCount = roomCount * maxLimit;
  const reducedRoomCount = roomCount - 1;
  const reducedMaxCount = reducedRoomCount * maxLimit;
  const balCount = adultsCount - maxLimit;

  const incrementRoomCounter = () => {
    setRoomCount(roomCount + 1);
    if (totalCount <= maxCount) {
      incrementAdultsCounter();
    }
  };

  const decrementRoomCounter = () => {
    setRoomCount(roomCount - 1);
    if (totalCount > reducedMaxCount) {
      let count = totalCount - reducedMaxCount;
      if (childrenCount > count) {
        setChildrenCount(childrenCount - count);
      } else if (reducedRoomCount !== minLimit) {
        let limit = maxLimit - childrenCount;
        setChildrenCount(0);
        setAdultsCount(adultsCount - limit);
      } else {
        setChildrenCount(0);
        setAdultsCount(adultsCount - balCount);
      }
    }
  };

  const incrementAdultsCounter = () => {
    setAdultsCount(adultsCount + 1);
  };

  const decrementAdultsCounter = () => {
    if (totalCount > roomCount) {
      setAdultsCount(adultsCount - 1);
    }
  };

  const incrementChildrenCounter = () => {
    setChildrenCount(childrenCount + 1);
  };

  const decrementChildrenCounter = () => {
    setChildrenCount(childrenCount - 1);
  };

  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <FontAwesomeIcon icon={faUsers} style={styles.icon} />
        <Text style={styles.title}>
          Choose number of
          <Text style={styles.highlight}> people</Text>
        </Text>
      </View>
      <View style={styles.boxContainer}>
        <View style={[styles.tableContainer, styles.bottomBorder]}>
          <Header name="ROOMS" source={faBed} />
          <Counter
            count={roomCount}
            decrementDisabled={roomCount === minLimit ? true : false}
            incrementDisabled={roomCount >= maxRoomLimit ? true : false}
            onDecrementPressed={decrementRoomCounter}
            onIncrementPressed={incrementRoomCounter}
          />
        </View>
        <View style={[styles.tableContainer, styles.bottomBorder]}>
          <Header name="ADULTS" source={faUser} />
          <Counter
            count={adultsCount}
            decrementDisabled={adultsCount === minLimit ? true : false}
            incrementDisabled={totalCount >= maxCount ? true : false}
            onDecrementPressed={decrementAdultsCounter}
            onIncrementPressed={incrementAdultsCounter}
          />
        </View>
        <View style={styles.tableContainer}>
          <Header name="CHILDREN" source={faChild} />
          <Counter
            count={childrenCount}
            decrementDisabled={childrenCount === 0 ? true : false}
            incrementDisabled={totalCount >= maxCount ? true : false}
            onDecrementPressed={decrementChildrenCounter}
            onIncrementPressed={incrementChildrenCounter}
          />
        </View>
      </View>
    </View>
  );
};

export default Home;
