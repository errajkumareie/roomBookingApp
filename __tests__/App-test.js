/**
 * @format
 */

import 'react-native';
import React from 'react';
import Home from '../src/screens/Home';
import Header from '../src/components/Header';
import Counter from '../src/components/Counter';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

it('renders Home Correctly', () => {
  const home = renderer.create(<Home />).toJSON();
  expect(home).toMatchSnapshot();
});

it('renders Header Correctly', () => {
  const header = renderer.create(<Header />).toJSON();
  expect(header).toMatchSnapshot();
});

it('renders Counter Correctly', () => {
  const counter = renderer.create(<Counter />).toJSON();
  expect(counter).toMatchSnapshot();
});
