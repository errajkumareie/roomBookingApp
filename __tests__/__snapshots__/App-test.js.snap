// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Counter Correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingRight": 20,
      "paddingVertical": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f056",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
          ],
          "iconName": "minus-circle",
          "prefix": "fas",
        }
      }
      style={
        Object {
          "color": "navy",
          "padding": 15,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
        "justifyContent": "flex-start",
        "marginHorizontal": 20,
        "marginTop": 2,
        "width": 28,
      }
    }
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f055",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
          ],
          "iconName": "plus-circle",
          "prefix": "fas",
        }
      }
      style={
        Object {
          "color": "mediumvioletred",
          "marginLeft": -10,
          "padding": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`renders Header Correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingRight": 20,
      "paddingVertical": 20,
    }
  }
>
  <
    style={
      Object {
        "color": "navy",
        "padding": 15,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "navy",
        "fontSize": 18,
        "justifyContent": "flex-start",
        "marginLeft": 15,
        "marginTop": 5,
      }
    }
  />
</View>
`;

exports[`renders Home Correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "height": 100,
      "marginTop": 32,
      "padding": 10,
      "paddingHorizontal": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingRight": 20,
        "paddingVertical": 20,
      }
    }
  >
    <
      icon={
        Object {
          "icon": Array [
            640,
            512,
            Array [],
            "f0c0",
            "M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z",
          ],
          "iconName": "users",
          "prefix": "fas",
        }
      }
      style={
        Object {
          "color": "navy",
          "padding": 15,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "navy",
          "fontSize": 20,
          "marginLeft": 15,
        }
      }
    >
      Choose number of
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
         people
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "borderWidth": 1,
        "flexDirection": "column",
        "height": 220,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 20,
          },
          Object {
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f236",
                "M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z",
              ],
              "iconName": "bed",
              "prefix": "fas",
            }
          }
          style={
            Object {
              "color": "navy",
              "padding": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "navy",
              "fontSize": 18,
              "justifyContent": "flex-start",
              "marginLeft": 15,
              "marginTop": 5,
            }
          }
        >
          ROOMS
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f056",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
                ],
                "iconName": "minus-circle",
                "prefix": "fas",
              }
            }
            style={
              Object {
                "color": "navy",
                "padding": 15,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "fontSize": 22,
              "justifyContent": "flex-start",
              "marginHorizontal": 20,
              "marginTop": 2,
              "width": 28,
            }
          }
        >
          1
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f055",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
                ],
                "iconName": "plus-circle",
                "prefix": "fas",
              }
            }
            style={
              Object {
                "color": "mediumvioletred",
                "marginLeft": -10,
                "padding": 15,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 20,
          },
          Object {
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f007",
                "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
              ],
              "iconName": "user",
              "prefix": "fas",
            }
          }
          style={
            Object {
              "color": "navy",
              "padding": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "navy",
              "fontSize": 18,
              "justifyContent": "flex-start",
              "marginLeft": 15,
              "marginTop": 5,
            }
          }
        >
          ADULTS
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f056",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
                ],
                "iconName": "minus-circle",
                "prefix": "fas",
              }
            }
            style={
              Object {
                "color": "navy",
                "padding": 15,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "fontSize": 22,
              "justifyContent": "flex-start",
              "marginHorizontal": 20,
              "marginTop": 2,
              "width": 28,
            }
          }
        >
          1
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f055",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
                ],
                "iconName": "plus-circle",
                "prefix": "fas",
              }
            }
            style={
              Object {
                "color": "mediumvioletred",
                "marginLeft": -10,
                "padding": 15,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <
          icon={
            Object {
              "icon": Array [
                384,
                512,
                Array [],
                "f1ae",
                "M120 72c0-39.765 32.235-72 72-72s72 32.235 72 72c0 39.764-32.235 72-72 72s-72-32.236-72-72zm254.627 1.373c-12.496-12.497-32.758-12.497-45.254 0L242.745 160H141.254L54.627 73.373c-12.496-12.497-32.758-12.497-45.254 0-12.497 12.497-12.497 32.758 0 45.255L104 213.254V480c0 17.673 14.327 32 32 32h16c17.673 0 32-14.327 32-32V368h16v112c0 17.673 14.327 32 32 32h16c17.673 0 32-14.327 32-32V213.254l94.627-94.627c12.497-12.497 12.497-32.757 0-45.254z",
              ],
              "iconName": "child",
              "prefix": "fas",
            }
          }
          style={
            Object {
              "color": "navy",
              "padding": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "navy",
              "fontSize": 18,
              "justifyContent": "flex-start",
              "marginLeft": 15,
              "marginTop": 5,
            }
          }
        >
          CHILDREN
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f056",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
                ],
                "iconName": "minus-circle",
                "prefix": "fas",
              }
            }
            style={
              Object {
                "color": "navy",
                "padding": 15,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "fontSize": 22,
              "justifyContent": "flex-start",
              "marginHorizontal": 20,
              "marginTop": 2,
              "width": 28,
            }
          }
        >
          0
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f055",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
                ],
                "iconName": "plus-circle",
                "prefix": "fas",
              }
            }
            style={
              Object {
                "color": "mediumvioletred",
                "marginLeft": -10,
                "padding": 15,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
